`make compile` fails in pristine clone of repo

On the `develop` branch, a `make compile` in a pristine clone results in:

```
12:33 $ make
escript get_comp_opts.escript
make  erlc-lfec
make[1]: Entering directory '/home/mjl/work/castlerock/src/test/lfe'
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_shell.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_codegen.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_io_pretty.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_macro.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_io_write.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_lint.erl
src/lfe_lint.erl:1477: Warning: function add_errors/3 is unused
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_doc.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_io_format.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_comp.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_ms.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_env.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_macro_record.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_internal.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_pmod.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_macro_include.erl
src/lfe_macro_include.erl:306: Warning: variable 'Line' is unused
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_edlin_expand.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfescript.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_parse.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_bits.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_lib.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_macro_export.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_gen.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_io.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_qlc.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_trans.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_types.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_init.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_eval.erl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe.erl
erlc -o src src/lfe_scan.xrl
erlc -I include -o ebin -DERLANG_VERSION=\"19.3\" -DHAS_MAPS=true -DHAS_FULL_KEYS=true -DNEW_REC_CORE=true -DNEW_RAND=true -W1 src/lfe_scan.erl
install -m755 bin/lfeexec /usr/local/lib/lfe/bin
install: cannot stat 'bin/lfeexec': No such file or directory
Makefile:121: recipe for target 'bin/lfeexec' failed
make[1]: *** [bin/lfeexec] Error 1
rm src/lfe_scan.erl
make[1]: Leaving directory '/home/mjl/work/castlerock/src/test/lfe'
Makefile:95: recipe for target 'compile' failed
make: *** [compile] Error 2
```

Switching to `master`, the build succeeds. Switching back to `develop`, it is now possible to build.

I'm guessing this is because some things in `bin` (`lfeexec`?) gets built properly on `master`, and then becomes available to subsequent builds on `develop`. As the initial build on `develop` also seems to try to do an install, I'm guessing the whole build process gets derailed somehow, possibly because of the missing `lfeexec`.

For reference, I'm doing this on Ubuntu 17.04 64bit, using Erlang Systems Erlang 19.3.

--- a/Makefile
+++ b/Makefile
@@ -89,7 +89,7 @@ $(EBINDIR)/%.beam: $(LSRCDIR)/%.lfe
 
 all: compile
 
-.PHONY: compile erlc-compile lfec-compile erlc-lfec emacs install docs clean docker-build docker-push docker update-mandb
+.PHONY: compile erlc-compile lfec-compile erlc-lfec emacs install install-beam install-bin install-man docs clean docker-build docker-push docker update-mandb
 
 compile: comp_opts.mk
 	$(MAKE) $(MFLAGS) erlc-lfec
@@ -117,17 +117,23 @@ comp_opts.mk:
 
 -include comp_opts.mk
 
-$(BINDIR)/lfe%:
-	$(INSTALL_BIN) $@ $(DESTBINDIR)
+install: compile install-beam install-bin install-man
 
-install: compile install-man
+install-beam:
 	rm -Rf $(DESTEBINDIR)
 	$(INSTALL_DIR) $(DESTEBINDIR)
-	$(INSTALL_DATA) $(EBINDIR)/$(APP_DEF) $(DESTEBINDIR)
-	$(INSTALL_DATA) $(addprefix $(EBINDIR)/, $(EBINS)) $(DESTEBINDIR)
-	$(INSTALL_DATA) $(addprefix $(EBINDIR)/, $(LBINS)) $(DESTEBINDIR)
+	$(INSTALL_DATA) -t $(DESTEBINDIR) \
+		$(EBINDIR)/$(APP_DEF) \
+		$(addprefix $(EBINDIR)/, $(EBINS)) \
+		$(addprefix $(EBINDIR)/, $(LBINS))
+
+install-bin:
 	$(INSTALL_DIR) $(DESTBINDIR)
-	$(MAKE) $(BINDIR)/lfe $(BINDIR)/lfec $(BINDIR)/lfedoc $(BINDIR)/lfescript
+	$(INSTALL_BIN) -t $(DESTBINDIR) \
+		$(BINDIR)/lfe \
+		$(BINDIR)/lfec \
+		$(BINDIR)/lfedoc \
+		$(BINDIR)/lfescript
 	ln -sf $(DESTBINDIR)/* $(PREFIX)/bin/
 
 clean:

